pillow_extension.py

```
@@ -0,0 +1,64 @@
+        position_text (tuple): Position of text.
+        text_origin (str): Text to highlight.
+        text_align (str): Text alignment.
+        text_color (tuple): Color of text.
+        highlighting_color (tuple): Color of highlighting.
+
+    """
+    try:
+        image = Image.open(image_src)
+    except FileNotFoundError:
+        logging.error(f"Image file '{image_src}' not found.")
+        raise ImageProcessingError(f"Failed to load image: {image_src}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)